### PR TITLE
[libc]Add proxy header hdr/float-macros.h

### LINKED ISSUE #88433: [libc] Add proxy header `hdr/float-macros.h`
Context: https://github.com/llvm/llvm-project/pull/87017 

Similar PR: https://github.com/llvm/llvm-project/pull/87598, https://github.com/llvm/llvm-project/issues/87863.

Add a proxy header for `float.h` so that our implementations and tests will work nicely for both overlay and full build modes.

- [ ] Add `libc/hdr/float_macros.h` that will include the system <float.h> in overlay mode or include `libc/include/llvm-libc-macros/float-macros.h` in full build mode.
- [ ] Add its corresponding cmake targets in `libc/hdr/CMakeLists.txt`.
- [ ] Remove the `#include_next <float.h>` in `libc/include/llvm-libc-macros/float-macros.h` and all the diagnostic suppression around it.
- [ ] Move the *filling-in* parts such as
```
#ifndef FLT_RADIX
#define FLT_RADIX __FLT_RADIX__
#endif // FLT_RADIX
```
to the proxy header `libc/hdr/float_macros.h` in the overlay mode, after including the system `<float.h>`.
- [ ] Change macro definitions in the full build header `libc/include/llvm-libc-macros/float-macros.h` to something like:
```
#ifdef __FLT_RADIX__
#define FLT_RADIX __FLT_RADIX__
#else
#define FLT_RADIX 2
#endif // FLT_RADIX
```
- [ ] Change all the usage of `include/llvm-libc-macros/float-macros.h` to `hdr/float_macros.h`, and update the CMake dependency accordingly.
- [ ] Add their corresponding bazel targets in utils/bazel/llvm-project-overlay/libc/BUILD.bazel.
- [ ] Update bazel dependencies.
- [ ] Test in overlay mode:
```
from "llvm-project/build":
$ cmake ../llvm -GNinja -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS=libc
$ ninja libc
$ ninja check-libc
```
- [ ] Test in full build mode:
```
from "llvm-project/build":
$ cmake ../llvm -GNinja -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS="llvm;clang;compiler-rt;libc" -DLLVM_LIBC_INCLUDE_SCUDO=ON -DLLVM_LIBC_FULL_BUILD=ON
$ ninja libc
$ ninja check-libc
```
- [ ] Test bazel build:
```
from "llvm-project/utils/bazel/"
$ bazel clean
$ bazel build --config=generic_clang --features=layering_check @llvm-project//libc/...
$ bazel test --config=generic_clang --features=layering_check @llvm-project//libc/...
```

### PR DESCRIPTION
**Pull request by aniplcc**
Closes #88433

### CODE REVIEW

**Comment by lntue** [comment]
@aniplcc Sorry I just got back from vacation.  Do you mind syncing the PR to the current HEAD?  Thanks,

File: libc/include/llvm-libc-macros/float-macros.h [hunk 1]
```diff
@@ -9,163 +9,206 @@
...
-
-#ifndef FLT_RADIX
+#ifdef __FLT_RADIX__
 #define FLT_RADIX __FLT_RADIX__
+#else 
+#define FLT_RADIX 2
 #endif // FLT_RADIX
 
-#ifndef FLT_EVAL_METHOD
+#ifdef FLT_EVAL_METHOD
```
Comment (lntue) [inline libc/include/llvm-libc-macros/float-macros.h:18]: This should be `#ifdef __FLT_EVAL_METHOD__`.  Update similar checks in this file.

File: libc/include/llvm-libc-macros/float-macros.h [hunk 2]
```diff
@@ -9,163 +9,206 @@
...
 #define LDBL_MANT_DIG __LDBL_MANT_DIG__
+#else 
+#define LDBL_MANT_DIG 113
 #endif // LDBL_MANT_DIG
 
-#ifndef FLT_MIN
+#ifdef FLT_MIN
 #define FLT_MIN __FLT_MIN__
+#else 
+#define FLT_MIN 1E-37
```
Comment (lntue) [inline libc/include/llvm-libc-macros/float-macros.h:85]: use hexadecimal constants instead: `#define FLT_MIN 0x1.0p-126f`

File: libc/include/llvm-libc-macros/float-macros.h [hunk 3]
```diff
@@ -9,163 +9,206 @@
...
 #endif // FLT_MIN
 
-#ifndef DBL_MIN
+#ifdef DBL_MIN
 #define DBL_MIN __DBL_MIN__
+#else 
+#define DBL_MIN 1E-37
```
Comment (lntue) [inline libc/include/llvm-libc-macros/float-macros.h:91]: `0x1.0p-1022`

File: libc/include/llvm-libc-macros/float-macros.h [hunk 4]
```diff
@@ -9,163 +9,206 @@
...
 #endif // DBL_MIN
 
-#ifndef LDBL_MIN
+#ifdef LDBL_MIN
 #define LDBL_MIN __LDBL_MIN__
+#else 
+#define LDBL_MIN 1E-37
```
Comment (lntue) [inline libc/include/llvm-libc-macros/float-macros.h:97]: This value will depend on CPU architecture + OS.

**Comment by aniplcc** [comment]
apologies for the delay, busy with finals atm. Will update arch dependent ldbl constants soon.

**Pull request closed without merging**